python/src/scipp/typing.py

```diff
@@ -49,3 +49,6 @@ def has_numeric_type(obj: _std_typing.Any) -> bool:
 
 #: Any object that behaves like a scipp.Dataset in most operations.
 DatasetLike = _std_typing.Union[DataArrayLike, sc.Dataset]
```

Comment:
Unify `DatasetLike` and `DataArrayLike` into `LabelledArrayLike`?

Reply:
👍

Reply:
I'm not sure `LabeledArray` is the best option here, as to me it would suggest that there is only one array, so it would be for a `DataArray`. It feels like it doesn't fit `Dataset` so well...